File: Unit-3-Inheritance/Class-Examples/Vehicle.java
public class Vehicle {
  //properties
  private String color;
  private int wheels;

  //getters
  public String getColors () {
    return this.color;
  }
  public int getWheels() {
    return this.wheels;
  }

  //setters
  public String setColor(String color) {
    return this.color = color;
  }
  public int setWheels(int wheels) {
    return this.wheels = wheels;
  }

  //methods
  public String honk () {
    return "Honk!";
  }
  public String output () {
    return "This car has " + this.wheels + " wheels and is " + this.color + ".";
  }

  //main
  public static void main(String [] args) {
    Vehicle aaron = new Vehicle();
    System.out.println(aaron.honk());
    aaron.setColor("silver");
    aaron.setWheels(4);
    System.out.println(aaron.output());
  }
}

File: Unit-3-Inheritance/Class-Examples/vehicle.js
//Constuctor

var Vehicle = function(wheels, color) {
  //properties
  this.wheels = wheels;
  this.color = color;
};

Vehicle.prototype.honk = function () {
  return 'Honk!';
};

Vehicle.prototype.output = function () {
  return 'This car has ' + this.wheels + ' wheels and is ' + this.color + '.';
};

// instances
var cliff = new Vehicle(4, 'red');
console.log(cliff.honk());
console.log(cliff.output());

//Constructor
var Car = function(color) {
  this.wheels = 4;
  this.color = color;
}

//methods
Car.prototype = new Vehicle();
Car.prototype.drive = function (miles) {
  return 'This car just drove ' + miles + ' miles.';
}

//instances
var herbie = new Car('white');
console.log(herbie.honk());
console.log(herbie.output());
console.log(herbie.drive(55));